include/dxc/Support/microcom.h

```diff
@@ -101,7 +101,7 @@ inline T *CreateOnMalloc(IMalloc * pMalloc, Args&&... args) {
 
 template<typename T>
 void DxcCallDestructor(T *obj) {
-  obj->~T();
+  obj->T::~T();
```

Comment:
Disabling that warning in this way would have been fine with the `final` notion, but theoretically it should be possible to have subclasses of subclasses with virtual destructors beyond that [^1]? This would then call the wrong destructor, though it wasn't removed in your commit?

For my understanding, as this has all been long ago and I don't write a lot of C++: calling `operator delete(this)` directly is different in that it doesn't perform a virtual dispatch to any of the destructor operators (since that's done already by `DxcCallDestructor`) but merely frees the memory allocated by `new`?

[^1]: As long as it isn't intermixed with the pure-virtual interfaces that would otherwise get a skewed vtable.

Reply:
Perhaps this is fine after all, given that:

- COM objects should only be freed through the `Release` virtual function;
- Hence, every descendant should override it;
- Which will call the "lowest" destructor through `DxcCallDestructor`: it's known statically here as `T`, hence doesn't need to be virtual.